From 575a2850cbe6564f84d5cb43bae8a3191503e306 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Tue, 27 Feb 2018 20:00:56 +0100
Subject: [PATCH 1/5] Revert "klte-common: HAXX: "Fix" race condition in init"

* A better way has been found

This reverts commit 31509201c1dc83cb4db5aa759f042459e87878b4.

Change-Id: I4b1e9bdf75c48add769879675eedf1d5ed472a8f
---
 BoardConfigCommon.mk     |  2 +-
 rootdir/etc/init.qcom.rc | 18 +++++++-----------
 2 files changed, 8 insertions(+), 12 deletions(-)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index 721032c..450c8ef 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -45,7 +45,7 @@ DEVICE_MANIFEST_FILE += $(COMMON_PATH)/manifest.xml
 
 # Kernel
 BOARD_KERNEL_BASE := 0x00000000
-BOARD_KERNEL_CMDLINE := console=null androidboot.hardware=qcom user_debug=31 msm_rtb.filter=0x37 ehci-hcd.park=3 zcache.enabled=1 zcache.compressor=lz4 maxcpus=1
+BOARD_KERNEL_CMDLINE := console=null androidboot.hardware=qcom user_debug=31 msm_rtb.filter=0x37 ehci-hcd.park=3 zcache.enabled=1 zcache.compressor=lz4
 BOARD_KERNEL_IMAGE_NAME := zImage
 BOARD_KERNEL_PAGESIZE := 2048
 BOARD_KERNEL_SEPARATED_DT := true
diff --git a/rootdir/etc/init.qcom.rc b/rootdir/etc/init.qcom.rc
index 8a6a20b..49a3f69 100644
--- a/rootdir/etc/init.qcom.rc
+++ b/rootdir/etc/init.qcom.rc
@@ -965,17 +965,6 @@ on property:sys.boot_completed=1
     write /sys/module/msm_pm/modes/cpu3/retention/suspend_enabled 1
     write /sys/module/msm_thermal/core_control/enabled 0
 
-    # Bring all CPUs online, and set node permissions
-    write /sys/devices/system/cpu/cpu1/online 1
-    write /sys/devices/system/cpu/cpu2/online 1
-    write /sys/devices/system/cpu/cpu3/online 1
-    chmod 664 /sys/devices/system/cpu/cpu1/online
-    chmod 664 /sys/devices/system/cpu/cpu2/online
-    chmod 664 /sys/devices/system/cpu/cpu3/online
-    chown root system /sys/devices/system/cpu/cpu1/online
-    chown root system /sys/devices/system/cpu/cpu2/online
-    chown root system /sys/devices/system/cpu/cpu3/online
-
     # Configure the CPU governor
     write /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor interactive
     write /sys/devices/system/cpu/cpu1/cpufreq/scaling_governor interactive
@@ -994,6 +983,13 @@ on property:sys.boot_completed=1
     write /sys/devices/system/cpu/cpu2/cpufreq/scaling_min_freq 300000
     write /sys/devices/system/cpu/cpu3/cpufreq/scaling_min_freq 300000
 
+    chown root system /sys/devices/system/cpu/cpu1/online
+    chown root system /sys/devices/system/cpu/cpu2/online
+    chown root system /sys/devices/system/cpu/cpu3/online
+    chmod 664 /sys/devices/system/cpu/cpu1/online
+    chmod 664 /sys/devices/system/cpu/cpu2/online
+    chmod 664 /sys/devices/system/cpu/cpu3/online
+
     write /sys/module/msm_thermal/core_control/enabled 1
 
     # Configure the cpu-boost module
-- 
2.7.4

